ui_framework/components/modal/confirm_modal.js

```diff
@@ -9,57 +9,98 @@ import { KuiModalBody } from './modal_body';
 import { KuiModalBodyText } from './modal_body_text';
 import { KuiButton } from '../index';
 
-export function KuiConfirmModal({
-    message,
-    title,
-    onCancel,
-    onConfirm,
-    cancelButtonText,
-    confirmButtonText,
-    className,
-    ...rest }) {
-  const ariaLabel = rest['aria-label'];
-  const dataTestSubj = rest['data-test-subj'];
-  return (
-    <KuiModal
-      style={{ 'width': '450px' }}
-      data-tests-subj={ dataTestSubj }
-      aria-label={ ariaLabel }
-      className={ className }
-    >
-      {
-        title ?
-          <KuiModalHeader>
-            <KuiModalHeaderTitle data-test-subj="confirmModalTitleText">
-              { title }
-            </KuiModalHeaderTitle>
-          </KuiModalHeader>
-          : null
-      }
-      <KuiModalBody>
-        <KuiModalBodyText data-test-subj="confirmModalBodyText">
-          { message }
-        </KuiModalBodyText>
-      </KuiModalBody>
+export const CONFIRM_BUTTON = 'confirm';
+export const CANCEL_BUTTON = 'cancel';
 
-      <KuiModalFooter>
-        <KuiButton
-          type="hollow"
-          data-test-subj="confirmModalCancelButton"
-          onClick={ onCancel }
-        >
-          {cancelButtonText}
-        </KuiButton>
-        <KuiButton
-          type="primary"
-          data-test-subj="confirmModalConfirmButton"
-          onClick={ onConfirm }
-        >
-          {confirmButtonText}
-        </KuiButton>
-      </KuiModalFooter>
-    </KuiModal>
-  );
+export const CONFIRM_MODAL_BUTTONS = [
+  CONFIRM_BUTTON,
+  CANCEL_BUTTON,
+];
+
+export class KuiConfirmModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.confirmButton;
+    this.cancelButton;
+  }
+
+  componentDidMount() {
+    switch (this.props.defaultFocusedButton) {
+      case CANCEL_BUTTON:
+        this.cancelButton.focus();
+        break;
+      default:
+        this.confirmButton.focus();
+        break;
+    }
+  }
+
+  onKeyDown = (event) => {
+    // Treat the 'esc' key as a cancel indicator.
+    if (event.keyCode === 27) {
```

Comment:
I know this isn't part of this PR, but how do you feel about making the `27` here a constant somewhere? I see it used multiple times in the code so it might be nice for it to be defined once and then referred to as `ESCAPE_KEY_CODE` or something like that.